### PR TITLE
Automation/6261-chanel-qa-approval-before-pr-merge 

### PR DESCRIPTION
**Pull request by IsraelleHub**
## Description of Change
This should be a Github actions workflow that:

Runs on each PR Open
Runs on each PR "approval"
Only runs when merges are going into develop
Only runs on code that requires both FE and QA review
Also:

It should run immediately with a failure, blocking merged for code QA needs to review
On each approval, it should rerun to test that there are at least 3 approvals, one of which is from a member of the QA team ([flagship-mobile-qa](https://github.com/orgs/department-of-veterans-affairs/teams/flagship-mobile-qa/members)). Ideally, if we can check team membership that would be best, but if not having users listed directly is fine.
the [CODEOWNERS](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/.github/CODEOWNERS) file outlines areas where the code needs QA approval for merge
Include code that touches the app:

VAMobile/src
VAMobile/e2e
VAMobile/package.json
VAMobile/.detoxrc.json

Exclude PRs with only unit tests:
VAMobile/src/*.test.tsx
## Testing

- [ ] Tested on iOS 
- [ ] Tested on Android 

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Review by liztownd — CHANGES_REQUESTED** [review]
My gut response is that there needs to be a check to see if there are *any* reviews first, before checking to see if there are approvals.

**Review by liztownd — APPROVED** [review]

**Review by rbontrager — APPROVED** [review]

**Review by liztownd — CHANGES_REQUESTED** [review]
Unapproving. :D

**Review by rbontrager — DISMISSED** [review]

**Review by TKDickson — DISMISSED** [review]
approving as a test

**Review by TKDickson — DISMISSED** [review]
approval test again

**Review by rbontrager — DISMISSED** [review]

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (narin) [inline .github/workflows/pr-checks-for-qa-approvals.yml:63]: Should we be checking `$qa_team_members` instead of these hardcoded usernames? It doesn't seem like `$qa_team_members` is being used aside from the echo.

**Review by rbontrager — APPROVED** [review]

**Review by rbontrager — APPROVED** [review]

**Review by rbontrager — APPROVED** [review]

**Review by rbontrager — APPROVED** [review]

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (IsraelleHub) [reply .github/workflows/pr-checks-for-qa-approvals.yml:55]: we will be using hardcoded usernames. I removed $qateam_members

**Review by rbontrager — DISMISSED** [review]
Test approve

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (timwright12) [inline .github/workflows/pr-checks-for-qa-approvals.yml:22]: @IsraelleHub

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (timwright12) [reply .github/workflows/pr-checks-for-qa-approvals.yml:55]: Make sure this is noted somewhere in the QART documentation for onboarding/offboarding if we need to maintain this list. Can you spin up a ticket for post-MVP to use the team members in github (unless that's not possible)

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (timwright12) [inline .github/workflows/pr-checks-for-qa-approvals.yml:10]: What happens if QA approves before engineering? Should we clear off the QA approval?

Comment (IsraelleHub) [reply .github/workflows/pr-checks-for-qa-approvals.yml:10]: It will pass with a message 'this workflow has QA approval'

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (IsraelleHub) [reply .github/workflows/pr-checks-for-qa-approvals.yml:55]: I tried to use qa team members, but it was just giving me hard time. I can look into it deeper on another ticket as well.

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (narin) [reply .github/workflows/pr-checks-for-qa-approvals.yml:10]: @IsraelleHub I don't think this is the desired outcome. Typically we want 1 engineer to have approved the PR first before QA takes a look, so if we can add logic for it to only pass if there has also been 1 approval from @department-of-veterans-affairs/flagship-mobile-reviewers that would be ideal.

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (narin) [reply .github/workflows/pr-checks-for-qa-approvals.yml:22]: Please remove this trigger once you're done testing and this is ready for another round of review

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (theodur) [inline .github/workflows/pr-checks-for-qa-approvals.yml:19]: We'll probably want to include the `paths` option here as well, to prevent this from running on irrelevant PRs that have their head branch updated

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (IsraelleHub) [reply .github/workflows/pr-checks-for-qa-approvals.yml:55]: Ticket has been created for post mvp

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (IsraelleHub) [reply .github/workflows/pr-checks-for-qa-approvals.yml:19]: Are referring to these path theo?
 - 'VAMobile/src/**'
      - 'VAMobile/package.json'
      - '!VAMobile/src/**.test.*'
if yes, I included right before 'on pull_request_review"

**Review by TKDickson — APPROVED** [review]
approving as a test

**Review by TKDickson — DISMISSED** [review]
approval testing

**Review by Sparowhawk — APPROVED** [review]

**Review by Sparowhawk — APPROVED** [review]

**Review by Sparowhawk — DISMISSED** [review]

**Review by TKDickson — DISMISSED** [review]
testing approval again

**Review by theodur — DISMISSED** [review]

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (IsraelleHub) [reply .github/workflows/pr-checks-for-qa-approvals.yml:10]: Done. @narin please review

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (narin) [inline .github/workflows/pr-checks-for-qa-approvals.yml:69]: This `if` statement looks the same as the one on line 55. They can probably be combined.

**Review by narin — CHANGES_REQUESTED** [review]
Looking very close. Some minor suggestions and also there is a merge conflict.

**Review by rbontrager — DISMISSED** [review]

**Review by Sparowhawk — APPROVED** [review]

**Review by IsraelleHub — COMMENTED** [review]
testing

**Review by IsraelleHub — COMMENTED** [review]
testing

**Review by rbontrager — DISMISSED** [review]

**Review by Sparowhawk — DISMISSED** [review]

**Review by IsraelleHub — COMMENTED** [review]
testing4

**Review by IsraelleHub — COMMENTED** [review]
testing 5

**Review by Sparowhawk — DISMISSED** [review]

**Review by IsraelleHub — COMMENTED** [review]
testing6

**Review by IsraelleHub — COMMENTED** [review]
testing PR number fectching

**Review by IsraelleHub — COMMENTED** [review]
Check if pr number is fetched

**Review by IsraelleHub — COMMENTED** [review]
testing1

**Review by IsraelleHub — COMMENTED** [review]
verify team members are fetched correctly

**Review by rbontrager — APPROVED** [review]

**Review by IsraelleHub — COMMENTED** [review]
testing1

**Review by IsraelleHub — COMMENTED** [review]
testing2

**Review by IsraelleHub — COMMENTED** [review]
test1

**Review by IsraelleHub — COMMENTED** [review]
test2

**Review by IsraelleHub — COMMENTED** [review]
test3

**Review by IsraelleHub — COMMENTED** [review]
test3

**Review by IsraelleHub — COMMENTED** [review]
test4

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (theodur) [inline .github/workflows/pr-checks-for-qa-approvals.yml:26]: I realized this workflow will make QA's approval required for Dependabot PRs also, which those don't typically need QA approval to merge. Can we add logic to this job that makes it return successfully if the PR was opened by Dependabot? 

The easiest way to do this is probably to check for the author of the PR. This should be accessible via the `github` context:

```yml
github.event.pull_request_review.user.login == 'dependabot[bot]'
```

**Review by theodur — CHANGES_REQUESTED** [review]

**Review by narin — APPROVED** [review]

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (IsraelleHub) [reply .github/workflows/pr-checks-for-qa-approvals.yml:27]: @theodur please review whenever you can

**Review by theodur — APPROVED** [review]

**Comment by narin** [comment]
@rbontrager Can you please approve this PR so we can get this one merged?

**Pull request merged**
